File: CompHub/src/main/java/com/comphub/component/dto/ComponentShowcase.java
package com.comphub.component.dto;

import com.comphub.component.category.dto.CategoryDto;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.*;

import java.time.LocalDate;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ComponentShowcase {

    private Long id;
    private String name;
    private Set<CategoryDto> categories;
    private String imageUrl;
    private Creator creator;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate createdAt;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDate updatedAt;
    private String description;
}

File: CompHub-ui/src/app/features/frontend-component/models/component/component-showcase.ts
import {CategoryShowcase} from '../category/category-showcasse';
import {Creator} from '../user/creator';

export interface ComponentShowcase {
  id: number;
  name: string;
  categories: CategoryShowcase[];
  imageUrl: string;
  creator: Creator;
  createdAt: string;
  updatedAt: string;
  description: string;
}
